app/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py

```diff
@@ -340,6 +340,148 @@ def test_invalid_experiments_with_specific_languages(
 
         self.assertIn("languages", serializer.errors)
 
+    @parameterized.expand(
+        [
+            (
+                NimbusExperiment.Application.FOCUS_ANDROID,
+                NimbusExperiment.Version.FIREFOX_102,
+            ),
+            (
+                NimbusExperiment.Application.FENIX,
+                NimbusExperiment.Version.FIREFOX_102,
+            ),
+            (
+                NimbusExperiment.Application.IOS,
+                NimbusExperiment.Version.FIREFOX_101,
+            ),
+            (
+                NimbusExperiment.Application.FOCUS_IOS,
+                NimbusExperiment.Version.FIREFOX_101,
+            ),
+        ]
+    )
+    def test_valid_experiments_supporting_countries_versions(
+        self, application, firefox_version
+    ):
+
+        experiment_1 = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            application=application,
+            channel=NimbusExperiment.Channel.RELEASE,
+            firefox_min_version=firefox_version,
+        )
+        experiment_1.save()
+        serializer_1 = NimbusReviewSerializer(
+            experiment_1,
+            data=NimbusReviewSerializer(
+                experiment_1,
+                context={"user": self.user},
+            ).data,
+            context={"user": self.user},
+        )
+        self.assertTrue(serializer_1.is_valid())
```

Comment:
What is this case testing?  It seems like there's two tests in here, and one of them doesn't involve countries?

Reply:
It means the above-mentioned versions are valid, and we can test two scenarios based on that:

- When you select a specific country with valid versions
- When you don't select a specific country i.e. All countries with valid versions

Reply:
Right but I mean I expect a single test to have

1) Setup
2) Assertions

for a single case.  So if you're doing that cycle twice in a single experiment, when an assertion fails, you don't know which of the two cases it's trying to say are wrong, so if you want to keep both of these cases I'd suggest splitting them into two separate experiments 🙏

Reply:
sure sounds like a good plan 💃 🙏  thanks 😊

Reply:
Oh and if in one of your cases you specifically want to test the 'all countries' case, you'd have to explicitly specify `countries=[]` in the ExperimentFactory otherwise it will pick some at random:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/tests/factories.py#L321-L322

Reply:
got it all done, thank you @jaredlockhart 🙏